Framework/Algorithms/src/ResizeRectangularDetector.cpp

```diff
@@ -97,19 +97,19 @@ void ResizeRectangularDetector::exec() {
     throw std::runtime_error("Component with name " + ComponentName +
                              " was not found.");
 
-  auto &componentInfo = inputW->mutableComponentInfo();
-  const auto componentIndex = componentInfo.indexOf(comp->getComponentID());
-  if (!componentInfo.isDetector(componentIndex))
+  RectangularDetector_const_sptr det =
+      boost::dynamic_pointer_cast<const RectangularDetector>(comp);
```

Comment:
Yes agree. Maybe just recording the component indexes of the `RectangularDetectors` would be enough. That way a method could be provided that utilised something like `isDetector` and `parent(parent(index))` with the flag for the RectangularDetectors.